pkg/controllers/clusterregistrationtoken/handler.go

```diff
@@ -81,6 +86,25 @@ func (h *handler) OnChange(token *fleet.ClusterRegistrationToken, status fleet.C
 		if err != nil {
 			return nil, status, err
 		}
+	} else if len(sa.Secrets) == 0 {
+		secretCreated, err := secretutil.GetServiceAccountTokenSecret(sa, h.secretsController)
+		if err != nil {
+			return nil, status, err
+		}
+		secretReloaded, err := h.secretsCache.Get(token.Namespace, secretCreated.Name)
```

Comment:
`GetServiceAccountTokenSecret` waits for token and returns the secret, why do we need to reload or/and check of empty data token again?

Reply:
This is a check to ensure we can proceed with clusterRegistration workflow, if the secret is empty, registration workflow halts and needs a restart for fleet-controller to move forward. So making sure we proceed only token is well set.